src/Gui/MainWindow.cpp

```diff
@@ -150,11 +150,91 @@ class CustomMessageEvent : public QEvent
     int _timeout;
 };
 
+class DimensionWidget : public QPushButton, WindowParameter
+{
+    Q_OBJECT
+
+public:
+    DimensionWidget(QWidget* parent): QPushButton(parent), WindowParameter("Units")
+    {
+        setFlat(true);
+        setText(qApp->translate("Gui::MainWindow", "Dimension"));
+        setMinimumWidth(120);
+
+        //create the action buttons
+        auto* menu = new QMenu(this);
+        auto* actionGrp = new QActionGroup(menu);
+        int num = static_cast<int>(Base::UnitSystem::NumUnitSystemTypes);
+        for (int i = 0; i < num; i++) {
+            QAction* action = menu->addAction(QStringLiteral("UnitSchema%1").arg(i));
+            actionGrp->addAction(action);
+            action->setCheckable(true);
+            QObject::connect(action, &QAction::toggled, this, [this,i](bool checked) {
+                    if(checked) {
+                        // Set and save the Unit System
+                        Base::UnitsApi::setSchema(static_cast<Base::UnitSystem>(i));
+                        getWindowParameter()->SetInt("UserSchema", i);
+                        // Update the application to show the unit change
+                        Gui::Application::Instance->onUpdate();
+                    }
+                } );
+        }
+        setMenu(menu);
+        retranslateUi();
+        unitChanged();
+        getWindowParameter()->Attach(this);
+    }
+
+    ~DimensionWidget()
+    {
+        getWindowParameter()->Detach(this);
+    }
+
+    void OnChange(Base::Subject<const char*> &rCaller, const char * sReason) override
+    {
+        Q_UNUSED(rCaller)
+        if (strcmp(sReason, "UserSchema") == 0) unitChanged();
```

Comment:
I think the unitChanged() should be moved to the next line. Lint then also requires curly braces.

Reply:
Well, that was an attempt to keep the number of lines pretty low. :)
To me, this way is far less error-prone than the 2-line way with no braces -- BTW the linter doesn't say anything here --.
So if you require a multi-line form, I'll necessarily add the curly braces. ;)

Reply:
IIRC we once had this discussion to avoid to write this on the same line as the if condition because it's easier to overlook.

Reply:
Done